internal/controllers/cluster/cluster_controller_phases.go

```diff
@@ -61,6 +63,22 @@ func (r *Reconciler) reconcilePhase(_ context.Context, cluster *clusterv1.Cluste
 	if !cluster.DeletionTimestamp.IsZero() {
 		cluster.Status.SetTypedPhase(clusterv1.ClusterPhaseDeleting)
 	}
+
+	// Only record the event if the status has changed
+	if preReconcilePhase != cluster.Status.GetTypedPhase() {
+		// Failed clusters should get a Warning event
+		if cluster.Status.GetTypedPhase() == clusterv1.ClusterPhaseFailed {
+			var failureMessage string
```

Comment:
```suggestion
		failureMessage := pointer.StringDeref(cluster.Status.FailureMessage, "unknown")
```

Reply:
oh neat, didn't know that was possible. I love it!